### ISSUE TITLE
topicRedirect appears to process replies twice.

### ISSUE THREAD

**Issue by Lewwwk** [silentrob/superscript#236]
When running the following test, the function in the resulting reply gets triggered twice indicating that the reply is being processed twice. This has been leading to some unexpected behaviour.

```
+ test
- {keep}Testing redirect ^topicRedirect(__test_topic__, __test_reply__)

> topic __test_topic__
    + __test_reply__
    - Test reply ^get(name)
< topic
```
I used a console log in `./superscript/plugins/user.js -- get function` to help check for double trigger.

UPDATE: where I'm at and stuck, topicRedirect processes the full message, but then regular message processing takes over and does it again. So perhaps we force a stop in message processing after topic redirect is complete.

**Comment by Lewwwk** [comment]
I've confirmed it is happening with a bunch of logs but can't quite sort through why.  What I do see is that the reply and reply object are getting properly created in the first pass through and THEN the callback here: https://github.com/silentrob/superscript/blob/master/lib/reply/topicRedirect.js#L90-L93 calls back with the unprocessed reply which then gets processed again a second time.

I noticed this because one of my inline redirects in the reply had a topic flag, so it would process fine the first time but then it would give me strange results the SECOND time time through because the topic had been changed for the user and now it was generating replies from the new topic.

**Comment by Lewwwk** [comment]
This override is what gets rid of the first processed result, and leads to the second processing: https://github.com/silentrob/superscript/blob/master/lib/reply/topicRedirect.js#L71-L74

Without it though we get an error in `/superscript/lib/processtags.js:190`: 
```
          newReplyObject.id = getReplyObject.reply.id;
                                                  ^
TypeError: Cannot read property 'id' of undefined 
```

**Comment by silentrob** [comment]
I feel like what you are describing is part of the system architecture. SuperScript has a few recursive layers that leads to essentially the bot engine calling itself creating an internal loop. We do this when we redirect to a new topic, (topicRedirect, respond and inlineRedirect). In the logs you will also see a depth entry showing how many times we have gone deeper. 

When I run the test above I get the expected result

```
      bot.reply("user1", "test", function(err, reply) {
        reply.string.should.eql("Testing redirect Test reply");
        done();
      });
```
Also the final topic is `topicName: '__test_topic__'`

I added a keep to the reply and removed the get function. 
Lets verify that you are running the latest version of ss-parser I bumped it but did not bump superscript to reflect.

**Comment by Lewwwk** [comment]
Just updated and ran again - same result. So, as I understand it now, it is performing as expected. The replies themselves may be processed multiple times. It would also explain why {@trigger} don't exhaust the replies that they return. If they did, a reply would get exhausted time the reply was processed, resulting in more than one reply getting exhausted each time.

Where this is a particular issue for me is where my function in the reply changes user data each time it runs. So while the user gets one result, the reply has been processed multiple times and so the user record shows that the user got multiple results. I can't think of a way to catch that right now and only return the result once regardless of how many times the reply gets processed in generating a response to the user.

Looking back at some of my other issues, I suspect this may also be related to #220 #230 #244 #218 #194 and #171

**Comment by Lewwwk** [comment]
Have now also tried restructuring scripts to see if the same happens with `{@trigger}` and `^respond(topic).  In both cases the reply only gets processed once. This affords some opportunity for work arounds.

**Comment by silentrob** [comment]
Closing this issue as it should be fixed in v0.10.24
